src/main/java/teammates/common/datatransfer/questions/FeedbackRankRecipientsQuestionDetails.java

```diff
@@ -285,6 +310,60 @@ public String getQuestionResultStatisticsCsv(
         return normalisedRankOfResponse;
     }
 
+    /**
+     * Returns list of responses excluding responses given to self.
+     *
+     * @param responses a list of responses
+     * @return list of responses excluding self given responses
+     */
+    private List<FeedbackResponseAttributes> getResponsesExcludingSelf(List<FeedbackResponseAttributes> responses) {
+        List<FeedbackResponseAttributes> responsesExcludingSelf = new ArrayList<FeedbackResponseAttributes>();
+        for (FeedbackResponseAttributes response : responses) {
+            if (!response.giver.equalsIgnoreCase(response.recipient)) {
+                responsesExcludingSelf.add(response);
+            }
+        }
+        return responsesExcludingSelf;
+    }
+
+    /**
+     * Returns the average excluding self response text.
+     * Displays a dash if the user has only self response.
+     *
+     * @param df decimal format
+     * @param recipientRanksExcludingSelf list of recipient ranks excluding self response
+     * @param recipientKey recipient for which average is to be calculated
+     * @return average excluding self text
+     */
+    private String getAverageExcludingSelfText(DecimalFormat df,
+            Map<String, List<Integer>> recipientRanksExcludingSelf, String recipientKey) {
+        List<Integer> ranksExcludingSelf = recipientRanksExcludingSelf.get(recipientKey);
+        if (ranksExcludingSelf == null) {
+            return "-";
+        } else {
+            Double averageExcludingSelf = computeAverage(ranksExcludingSelf);
+            return df.format(averageExcludingSelf);
+        }
+    }
+
+    /**
+     * Returns map of recipient ranks excluding self if self ranks are to be excluded other wise return empty map.
```

Comment:
if we have to show "average excluding self" tab then it returns populated map otherwise just returns an empty map.